### ISSUE TITLE
0.17.0rc1: no effect when voting on a reply after its root post has been cashed out

### ISSUE THREAD

**Issue by abitmore** [steemit/steem#901]
According to [this code](https://github.com/steemit/steem/blob/v0.17.0rc1/libraries/chain/steem_evaluator.cpp#L1190):
```
void vote_evaluator::do_apply( const vote_operation& o )
{ try {
   database& _db = db();

   const auto& comment = _db.get_comment( o.author, o.permlink );

...

   if( _db.has_hardfork( STEEMIT_HARDFORK_0_12__177 ) && _db.calculate_discussion_payout_time( comment ) == fc::time_point_sec::maximum() )
   {
#ifndef CLEAR_VOTES

...

#endif
      return;
   }
```

`_db.calculate_discussion_payout_time( comment )` will return next cashout time of the root post, which is `fc::time_point_sec::maximum()` if the root post has been cashed out, so the vote will 
has no effect.

**Comment by mvandeberg** [comment]
Addressed in 	cd4f4ea

**Comment by mvandeberg** [comment]
Merged
